src/utils/NewsApi/NewsApi.service.js

```diff
@@ -0,0 +1,35 @@
+class NewsApi {
+    constructor({ url, key }) {
+        this.url = url;
+        this.key = key;
+        this.time = new Date();
+    }
+    async _connect(url) {
+        const res = await fetch(url);
+        if (res) {
```

Comment:
You need to check every response from the server if the response is `ok` or not

` if (res.ok) {` should be added


```js

    async _connect(url, headers) {
        const res = await fetch(url, headers);
        if (res.ok) {
            return res.json();
        }
        return await Promise.reject(`Error: ${res.status}`);
    }


```